### DISCUSSION TITLE
Help Wanted

### DISCUSSION THREAD

**Discussion by cseickel** [nvim-neo-tree/neo-tree.nvim#704] [Announcements]
The Neo-tree project could really use some awesome developers that have some free time and love that they can dedicate to maintaining the project. Up until recently, I have been the primary person to triage issues, fix bugs, plan new features, etc. We have had lots of great contributions from the community of course, which is very much appreciated, but I could really use someone to step up and take a lead role on the project. If we can get more than one person, even better!

I don't have the same amount of free time that I had when I started the project, and I am no longer really keeping up the way I should. I'm not looking to completely drop out of the project, but I would like to take a backseat if there's anyone out there interested in being a maintainer.

Of course, I won't just hand over the reins to anyone. There are a few people that have made significant contributions to the project that I would trust right away. Other plugin developers whose projects follow a similar mentality would also be welcomed and given authority quickly. Outside of that, I would want to be the final review on PRs and continue to play a major role in any architectural changes that are discussed for a while. New maintainers will be in an advisory role until I can get comfortable enough with them to give them PR approval access.

If anyone is interested, you can just reply here. If I don't get any takers, I'll repost this to Reddit in a few days.

**Comment by nyngwang** [comment]
Hi, I received your message. I will keep an eye on the Issues and try to help when possible.

**Reply by cseickel** [reply]
I knew you would @nyngwang! Thanks for sticking with neo-tree from the beginning, you've been a huge help in making neo-tree better.

**Comment by giovankes** [comment]
Hey! 

I saw your post on Reddit and will be sure to look out for issues and help where i can.

**Reply by cseickel** [reply]
Thanks, any help is appreciated.

**Comment by nhat-vo** [comment]
Hi! 

Thank you for making Neotree! It is a really amazing product and I'd love to help out!

**Comment by cseickel** [comment]
I probably should have listed out what type of help is needed so people could volunteer for specific things:

- [ ] Responding to issues (triage, how-to answers)
- [ ] Documenting (we really needed better documentation!)
- [ ] Bug fixes
- [ ] New Features
- [ ] Review PRs
- [ ] Plan and implement [version 3.0](https://github.com/nvim-neo-tree/neo-tree.nvim/issues/555)
- [ ] Lead the project into the future!

For anyone offering help, just let me know which of those things you are interested in.

Thanks!

**Reply by nhat-vo** [reply]
I would be interested in Responding to issues, Bug fixes, and New Features.

**Reply by cseickel** [reply]
Sounds great @nhat-vo! Go ahead and jump in on any issue you'd like to work on.

**Reply by nhat-vo** [reply]
Hi! Sorry I've been busy the last 2 weeks. I'm looking into #740, but I can't seem to set a label on issues. Could you enable this, please?

**Comment by miversen33** [comment]
I'm not looking to take on another project right now, but as I have dug pretty deep into the internals of Neo-tree and Nui, I can probably provide a bit of help on the maintenance aspect of Neo-tree :)

**Reply by cseickel** [reply]
Thanks Mike! It would be much appreciated.

**Comment by NavinShrinivas** [comment]
Hey there @cseickel, I would love to help in building and maintaining Neo tree, I don't have any in-depth experience building vim plugins, but would love to get started. Any good place to start helping?

I'm thinking of getting started by taking on a few bug fixes and later easing into new features.

**Reply by cseickel** [reply]
That sounds great @NavinShrinivas! If you're going to start working on something, just go ahead and drop a comment on the issue to claim it and maybe describe your proposed solution.

**Comment by ashkan117** [comment]
Hello @cseickel. I would love to help out as well. Can you suggest any issues that would be good first issue you would recommend starting with?

**Comment by miversen33** [comment]
Might be worth getting "good first issue" label and pruning the issues board to make this process easier? @cseickel

**Reply by cseickel** [reply]
Absolutely. I added the label, which is the easy part. What's not quite as easy is pruning the issues. I can take a shot at it later though.

**Reply by cseickel** [reply]
You know @miversen33 @ashkan117, the biggest chunk of work to do is just the triage of the issues themselves. It's hard to say which ones are good first issues because they all require conversation.

I would say that this triage is the most helpful thing anyone can do right now. In general, the issues break down like this:

- Probably half of the issues will turn out to be how-to and misunderstandings about how to configure and use Neo-tree (RTFM), 
- another 20% would be disagreement about how features are supposed to work that may just end with "let's agree to disagree",
- 20% are feature requests, which may be presented as bugs initially because of the above
- 10% are actual bugs, which are either recently introduced regressions or edge cases where the submitter needs to be prompted to provide more details to be able to recreate

So that's the first step of triage that is really needed. Once something has been positively identified as a bug or feature request, go ahead and claim it (you can just drop a comment on the issue) and start working on it.

BTW @miversen33 I've noticed you've started doing all of this, thanks!

**Reply by miversen33** [reply]
Ya I have had a bit of extra free time while I wait for a bug in luv to be addressed. So I figured I would make use of said free time :upside_down_face:

**Reply by ashkan117** [reply]
Going to be taking a look at #604 and give it a shot. This my first time with an open source project so things might go slowly but a simple enhancement sounds like a good starting point.

**Reply by miversen33** [reply]
@ashkan117 I documented the general way to tackle that issue, but if you have questions or need help, feel free to tag me. I may be slow to respond but I should be able to provide whatever help you need

**Reply by ashkan117** [reply]
@miversen33 Thanks! Sure I followed your suggestions so far and have some code that is close to solving it. I made a branch with the name of the github issue. Not sure what the process is from here though. Would I just push that branch to the remote repo?
